src/model/page-config-resolver.js

```diff
@@ -292,6 +335,120 @@ class MicrodataParser {
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      if (!content) {
+        continue;
+      }
+      if (content.toLowerCase() == 'true') {
+        value = true;
+      } else if (content.toLowerCase() == 'false') {
+        value = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return !value;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if a node is valid based on the following
+   * - child of an item of type 'NewsArticle'
+   * - not a child of an item of type 'Section'
+   * - not seen before, marked using the alreadySeen tag
+   * @param current the node to be verified
+   * @param root the parent to track up to
+   * @param alreadySeen used to tag already visited nodes
+   * @return {!boolean} valid node
+   * @private
+   */
+  isValidElement_(current, root, alreadySeen) {
+    while (current && !current[alreadySeen]) {
+      current[alreadySeen] = true;
+      if (current.hasAttribute('itemscope')) {
+        const type = current.getAttribute('itemtype');
+        if (type == 'http://schema.org/NewsArticle') {
+          // This parent may need to be check for other nodes
+          current[alreadySeen] = false;
+          return true;
+        } else if (type.indexOf('Section') >= 0) {
+          return false;
+        }
+      }
+      current = current.parentNode;
+    }
+    return false;
+  }
+
+  /**
+   * Obtains the product ID that meets the requirements
+   * - child of an item of type 'NewsArticle'
+   * - Not a child of an item of type 'Section'
+   * - child of an item of type 'productID'
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?string} product ID, if found
+   * @private
+   */
+  discoverProductId_(root) {
+    const ALREADY_SEEN = 'alreadySeenForProductInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='productID']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      const item = element.closest('[itemtype][itemscope]');
+      const type = item.getAttribute('itemtype');
+      if (type.indexOf('Product') <= -1) {
+        continue;
+      }
+      if (this.isValidElement_(item.parentNode, root, ALREADY_SEEN)) {
+        return content;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Extracts microdata embedded in the DOM
+   * @return {!Microdata} microdata found in the doc
+   */
+  tryExtractConfig_() {
+    let itemId = 0;
+    const results = [];
+    const nodeList = this.doc_.getRootNode().querySelectorAll(
+        "[itemscope][itemtype='http://schema.org/NewsArticle']");
+    const domReady = this.doc_.isReady();
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      if (!domReady && !hasNextNodeInDocumentOrder(element)) {
```

Comment:
Hmm. That'd likely block us for the remainder of parsing. Why can't we search within partial DOM?

Reply:
I derived these requirements from the tests in the JSON LD parser. Regardless of the parser, I assumed that those test cases imply similar requirements on Microdata parser. In this case if dom is not ready, it ensure if there's atleast one more node to process before proceeding? Is that not expected?

Reply:
Yeah, this is a very different case. The issue with LD parse is that it parses JSON text, which must be complete to be parseable, otherwise it completely fails. So, I used `hasNextNodeInDocumentOrder` as an indication that JSON text is fully available. We don't need this for microdata. As a matter of fact, I believe, this is one of its benefits. And you don't need `domReady` here either.

Reply:
Understood.